rewrite-java-tck/src/main/java/org/openrewrite/java/tree/UnnamedVariableTest.java

```diff
@@ -0,0 +1,337 @@
+          java(
+            """
+              class Test {
+                  String test(Object obj) {
+                      return switch (obj) {
+                          case Integer _ -> "It's an Integer";
+                          case Long _ -> "It's a Long";
+                          case Double _ -> "It's a Double";
+                          default -> "It's something else";
+                      };
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void unnamedPatternWithGuard() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  String test(Object obj) {
+                      return switch (obj) {
+                          case String _ when obj.toString().length() > 5 -> "Long string";
+                          case String _ -> "Short string";
+                          default -> "Not a string";
+                      };
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void unnamedVariableInRecordPattern() {
+        rewriteRun(
+          java(
+            """
+              record Point(int x, int y) {}
+
+              class Test {
+                  void test(Object obj) {
+                      if (obj instanceof Point(int x, _)) {
+                          System.out.println("x = " + x);
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void nestedUnnamedPatterns() {
+        rewriteRun(
+          java(
+            """
+              record Box<T>(T content) {}
+
+              class Test {
+                  void test(Object obj) {
+                      if (obj instanceof Box(Box(_))) {
+                          System.out.println("Nested box");
+                      }
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void unnamedVariableInLocalDeclaration() {
+        rewriteRun(
+          java(
+            """
+              import java.util.Queue;
+              import java.util.LinkedList;
+
+              class Test {
+                  void test() {
+                      Queue<String> queue = new LinkedList<>();
+                      queue.offer("first");
+                      queue.offer("second");
+
+                      var _ = queue.poll();
+                      var second = queue.poll();
+                      System.out.println(second);
+                  }
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    @Disabled
+    void unnamedVariableInAssignment() {
+        rewriteRun(
+          java(
+            """
+              class Test {
+                  void test() {
+                      int _;
+                      _ = 42;
+                      _ = 100;
+                  }
+              }
+              """
+          )
+        );
+    }
```

Comment:
I don't think this test makes sense, it doesn't look like reassigning a value is allowed

Reply:
Then let's remove it!